### ISSUE TITLE
Keys that start with a digit are translated to XX

### ISSUE THREAD

**Issue by TimLudwinski** [variety/variety#119]
For example, the following documents:

```
{ "_id" : ObjectId("5734a29add3389559ea1c828"), "1" : "test" }
{ "_id" : ObjectId("5734a2a8dd3389559ea1c829"), "1" : "test" }
{ "_id" : ObjectId("5734a2f42d14206bcade13e1"), "1" : { "2" : "test" } }
{ "_id" : ObjectId("5734a30a7446151d678b7a1d"), "1" : { "2asdf" : "test" } }
{ "_id" : ObjectId("5734a59221306216d413af16"), "1" : [ { "2asdf" : "test" } ] }
```
will produce the following output:

```
+------------------------------------------------------------------------+
| key         | types                           | occurrences | percents |
| ----------- | ------------------------------- | ----------- | -------- |
|           1 | Array (1),Object (2),String (2) |           5 |    100.0 |
| _id         | ObjectId                        |           5 |    100.0 |
| 1.XX.XXasdf | String                          |           1 |     20.0 |
| 1.XXasdf    | String                          |           1 |     20.0 |
+------------------------------------------------------------------------+
```

I have a fix for this as well as another new feature, but I need to finish my personal testing of this.

**Comment by todvora** [comment]
Hi, some of those XX are OK (see #29). But the regex is probably not correct and shouldn't translate keys like ```2asdf```, only unnamed arrays. 

I'll wait for your fix then :-)

Thanks,
Tomas

**Comment by todvora** [comment]
Also, when you are testing, check our [current test suite](https://github.com/variety/variety/tree/master/spec) & [test documentation](https://github.com/variety/variety#development-hacking). It would be cool, if you could add some tests too.

**Comment by JamesCropcho** [comment]
Hey there, good find with that bug. Any movement on this?

**Comment by todvora** [comment]
I've prepared a test to verify current incorrect behaviour: https://github.com/variety/variety/pull/121

It doesn't fix the issue itself and the PR doesn't touch ```variety.js``` file -> it should be easy to merge it with @TimLudwinski's changes.

**Comment by todvora** [comment]
Closing, already fixed by #122
